ReSharper.FSharp/src/FSharp.Psi/src/Impl/DeclaredElement/FSharpPropertyWithExplicitAccessors.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System.Collections.Generic;
+using JetBrains.ReSharper.Plugins.FSharp.Psi.Impl.Tree;
+using JetBrains.ReSharper.Plugins.FSharp.Psi.Tree;
+using JetBrains.ReSharper.Plugins.FSharp.Psi.Util;
+using JetBrains.ReSharper.Psi;
+
+namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Impl.DeclaredElement
+{
+  internal class FSharpPropertyWithExplicitAccessors : FSharpPropertyBase<MemberDeclaration>, IFSharpProperty
+  {
+    private readonly IAccessorDeclaration myGetter;
+    private readonly IAccessorDeclaration mySetter;
+
+    public FSharpPropertyWithExplicitAccessors(IMemberDeclaration declaration) : base(declaration)
+    {
+      var accessors = declaration.AccessorDeclarations;
+      myGetter = accessors.TryGet(AccessorKind.GETTER);
+      mySetter = accessors.TryGet(AccessorKind.SETTER);
+    }
+
+    public override bool IsReadable => myGetter != null;
+    public override bool IsWritable => mySetter != null;
+    public override IAccessor Getter => IsReadable ? myGetter.DeclaredElement as IAccessor : null;
+    public override IAccessor Setter => IsWritable ? mySetter.DeclaredElement as IAccessor : null;
+    public override AccessRights GetAccessRights() => AccessRights.PRIVATE;
+    public AccessRights RepresentationAccessRights => base.GetAccessRights();
+    public override IList<IParameter> Parameters => this.GetParameters(Mfv);
```

Comment:
Shouldn't these parameters belong to accessor?

Reply:
This is necessary in order for ReSharper to correctly display tooltips for explicit accessors in the future. When ReSharper reads such properties from the physical assembly, they are presented there with their arguments in a similar way.